otx/algorithms/detection/adapters/mmdet/models/detectors/custom_dino_detector.py

```diff
@@ -23,17 +23,97 @@ class CustomDINO(CustomDeformableDETR):
 
     def __init__(self, *args, task_adapt=None, **kwargs):
         super().__init__(*args, task_adapt=task_adapt, **kwargs)
-        self._register_load_state_dict_pre_hook(
-            self.load_state_dict_pre_hook,
-        )
 
-    @staticmethod
-    def load_state_dict_pre_hook(ckpt_dict, *args, **kwargs):
+        self.cls_layers.append("dn_query_generator.label_embedding.weight")
+
+    def load_state_dict_pre_hook(self, model_classes, ckpt_classes, ckpt_dict, *args, **kwargs):
         """Modify mmdet3.x version's weights before weight loading."""
 
         if list(ckpt_dict.keys())[0] == "level_embed":
             logger.info("----------------- CustomDINO.load_state_dict_pre_hook() called")
             # This ckpt_dict comes from mmdet3.x
+            ckpt_classes = [
```

Comment:
This is a list that will go away with the upgrade to mmdet 3.x, right?

Reply:
Yes.